src/compiler/checker.ts

```diff
@@ -3062,6 +3064,26 @@ namespace ts {
             setObjectTypeMembers(type, emptySymbols, callSignatures, constructSignatures, stringIndexType, numberIndexType);
         }
 
+        function intersectTypes(type1: Type, type2: Type): Type {
+            return !type1 ? type2 : !type2 ? type1 : getIntersectionType([type1, type2]);

```

Comment:
I'd really prefer this was split into

```TypeScript
if (!type1) {
    return type2;
}

if (!type2) {
    return type1;
}

return getIntersectionType([type1, type2]);
```

But if you're really set on keeping this as one statement, this might be easier on the eyes:

```TypeScript
return (!type1 && type2) ||
    (!type2 && type1) ||
    getIntersectionType([type1, type2]);
```